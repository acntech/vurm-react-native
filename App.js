// import React, { useState } from "react";
import GameScreen from "./src/screens/GameScreen";
import MainMenuScreen, {
  startGameLabel,
  instructionsLabel,
} from "./src/screens/MainMenuScreen";
import InstructionsScreen from "./src/screens/InstructionsScreen";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { GAME_TITLE } from "./src/constants";
import HighscoresScreen from "./src/screens/HighscoresScreen";
import { LogBox } from "react-native";

// firebase has yet to fix this, see https://github.com/firebase/firebase-js-sdk/issues/1847
LogBox.ignoreLogs(["AsyncStorage has been extracted from react-native core"]);

const Stack = createNativeStackNavigator();
const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MainMenu"
          component={MainMenuScreen}
          options={{ title: "Main Menu" }}
        />
        <Stack.Screen name={instructionsLabel} component={InstructionsScreen} />
        <Stack.Screen
          name={startGameLabel}
          component={GameScreen}
          options={{ title: GAME_TITLE, gestureEnabled: false }}
        />
        <Stack.Screen
          name={"Highscores"}
          component={HighscoresScreen}
          options={{ title: "Highscores" }}
        ></Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
