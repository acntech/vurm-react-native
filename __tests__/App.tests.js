import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import App from "../App";
import {
  startGameLabel,
  instructionsLabel,
} from "../src/screens/MainMenuScreen";

jest.useFakeTimers();
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe("<App />", () => {
  it("has 1 child", () => {
    // Arrange & Act
    const tree = render(<App />).toJSON();

    // Assert
    expect(tree.children.length).toBe(1);
  });
  it("renders correctly", () => {
    // Arrange & Act
    const tree = render(<App />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it("takes you to the game screen when clicking the Start Game button", () => {
    // Arrange
    const { getByText, getByTestId } = render(<App />);
    const toClick = getByText(startGameLabel);

    // Act
    fireEvent(toClick, "press");
    const game = getByTestId("Game");

    // Assert
    expect(game).toBeTruthy();
  });

  it("takes you to the instructions screen when clicking the Instructions button", () => {
    // Arrange
    const { getByText, getByTestId } = render(<App />);
    const toClick = getByText(instructionsLabel);

    // Act
    fireEvent(toClick, "press");
    const instructions = getByTestId("Instructions");

    // Assert
    expect(instructions).toBeTruthy();
  });
});
