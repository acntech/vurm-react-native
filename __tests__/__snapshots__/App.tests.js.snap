// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Main Menu"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 26,
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Vurm
          </Text>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "column",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "coral",
                  "borderRadius": 4,
                  "borderWidth": 0,
                  "marginBottom": 5,
                  "minWidth": "50%",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "fontWeight": "400",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "white",
                    },
                  ]
                }
              >
                Start Game
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "oldlace",
                  "borderRadius": 4,
                  "borderWidth": 0,
                  "marginBottom": 5,
                  "minWidth": "50%",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "fontWeight": "400",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "coral",
                    },
                  ]
                }
              >
                Instructions
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "oldlace",
                  "borderRadius": 4,
                  "borderWidth": 0,
                  "marginBottom": 5,
                  "minWidth": "50%",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "fontWeight": "400",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "coral",
                    },
                  ]
                }
              >
                Leaderboard
              </Text>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
